cli/lsp/diagnostics.rs

```diff
@@ -381,8 +392,8 @@ async fn generate_lint_diagnostics(
     for specifier in state_snapshot.documents.open_specifiers() {
       let version = state_snapshot.documents.version(specifier);
       let current_version = collection.get_version(specifier);
-      if version != current_version {
-        let media_type = MediaType::from(specifier);
+      let media_type = MediaType::from(specifier);
```

Comment:
This won't work for remote/data/blob files.  We need to get the media type from the document meta data and in other locations, for the sources meta data.

Reply:
Okay. I need understand a few things. Why do we need diagnostics for remote/data/blob files? Maybe I need to understand what are remote/data/blob files.

Reply:
Because if they are open in the editor, you want the editor to let you know if there are errors, right?  Even if they are open read only, having diagnostic information for them is pretty critical in understanding what is going on.

Reply:
Looks like we do not provide diagnostic information for remote files currently. I agree with your statement and I think we should have the feature as it's helpful, but I'm curious if the changes related to it can be done along with the PR that enables diagnostics for the remote files. WDYT?